src/Passes/Source/Rules/Tests/Unit/SelectOnOne.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Generators/DefaultProfileGenerator.hpp"
+#include "Rules/Factory.hpp"
+#include "Rules/ReplacementRule.hpp"
+#include "TestTools/IrManipulationTestHelper.hpp"
+#include "gtest/gtest.h"
+
+#include "Llvm/Llvm.hpp"
+
+#include <functional>
+
+using namespace microsoft::quantum;
+
+namespace
+{
+using IrManipulationTestHelperPtr = std::shared_ptr<IrManipulationTestHelper>;
+IrManipulationTestHelperPtr newIrManip(std::string const& script)
+{
+    IrManipulationTestHelperPtr ir_manip = std::make_shared<IrManipulationTestHelper>();
+
+    ir_manip->declareOpaque("Qubit");
+    ir_manip->declareOpaque("Result");
+
+    ir_manip->declareFunction("void @__quantum__qis__reset__body(%Qubit*)");
+    ir_manip->declareFunction("%Result* @__quantum__rt__result_get_one()");
+    ir_manip->declareFunction("i1 @__quantum__rt__result_equal(%Result*, %Result*)");
+    ir_manip->declareFunction("void @__quantum__qis__mz__body(%Qubit*, %Result*)");
+
+    if (!ir_manip->fromBodyString(script))
+    {
+        llvm::outs() << ir_manip->getErrorMessage() << "\n";
+        exit(-1);
+    }
```

Comment:
Looks like `ir_manip` is filled in with hard-coded data. The arg `script` is too. The failure of the call `ir_manip->fromBodyString(script)` feels like a detection of the internal bug in the code (not a detection of the error in the user-provided data). If that's always the case then I'd suggest:
```suggestion
    assert(ir_manip->fromBodyString(script));
```